server/events/policy_check_output_generator.go

```diff
@@ -0,0 +1,113 @@
+package events
+
+import (
+	"fmt"
+
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/lyft/feature"
+)
+
+const KeySeparator = "||"
+
+type PolicyCheckOutputStore struct {
+	Store map[string]*models.PolicyCheckSuccess
+}
+
+func NewPolicyCheckOutputStore() *PolicyCheckOutputStore {
+	return &PolicyCheckOutputStore{
+		Store: map[string]*models.PolicyCheckSuccess{},
+	}
+}
+
+func (p *PolicyCheckOutputStore) buildKey(projectName string, workspace string) string {
+	return fmt.Sprintf("%s%s%s", projectName, KeySeparator, workspace)
+}
+
+func (p *PolicyCheckOutputStore) GetOutputFor(projectName string, workspace string) *models.PolicyCheckSuccess {
+	key := p.buildKey(projectName, workspace)
+
+	if output, ok := p.Store[key]; ok {
+		return output
+	}
+	return nil
+}
+
+func (p *PolicyCheckOutputStore) WriteOutputFor(projectName string, workspace string, policyCheckResult command.ProjectResult) {
+	key := p.buildKey(projectName, workspace)
+
+	var output string
+	if policyCheckResult.Failure != "" {
+		output = policyCheckResult.Failure
+	} else if policyCheckResult.PolicyCheckSuccess != nil {
+		output = policyCheckResult.PolicyCheckSuccess.PolicyCheckOutput
+	}
+
+	val := &models.PolicyCheckSuccess{
+		PolicyCheckOutput: output,
+	}
+
+	p.Store[key] = val
+}
+
+type CommandOutputGenerator interface {
+	GenerateCommandOutput(ctx *command.Context, cmd *command.Comment) PolicyCheckOutputStore
+}
+
+type PolicyCheckCommandOutputGenerator struct {
+	PrjCommandRunner  ProjectPolicyCheckCommandRunner
+	PrjCommandBuilder ProjectPlanCommandBuilder
+	FeatureAllocator  feature.Allocator
+}
+
+func (f *PolicyCheckCommandOutputGenerator) GenerateCommandOutput(ctx *command.Context, cmd *command.Comment) PolicyCheckOutputStore {
+	if !f.isChecksEnabled(ctx) {
+		return PolicyCheckOutputStore{}
+	}
+
+	prjCmds, err := f.PrjCommandBuilder.BuildPlanCommands(ctx, &command.Comment{
+		RepoRelDir:  cmd.RepoRelDir,
+		Name:        command.Plan,
+		Workspace:   cmd.Workspace,
+		ProjectName: cmd.ProjectName,
+		LogLevel:    cmd.LogLevel,
+	})
+	if err != nil {
+		ctx.Log.WarnContext(ctx.RequestCtx, fmt.Sprintf("unable to build plan command: %s", err))
```

Comment:
why are we warning and then dropping the error here?

Reply:
Do we want to fail the operation if building plan command fails? 

From my understanding, if the build plan command fails which is not directly related to the `approve_policies` command, we should let the `approve_policies` operation go through similar to how we have it setup when using the commit status API without any dependency on `commandBuilder`. What are your thoughts on this?

Reply:
Then we would have no output on the PR once it's approved.  Doesn't sound like a great UX to me cuz we won't even know something is wrong potentially

Reply:
Hmm right, I've changed the `commandOuptutGenerator` to return error which is outputted into the checkrun if build plan fails.